projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts

```diff
@@ -166,6 +168,25 @@ describe('OrderApprovalDetailFormComponent', () => {
     assertSpinnerDisplayed();
   });
 
+  it('should display the back to list button when the approval does not require a decision.', () => {
+    getOrderApproval$.next({
+      ...mockOrderApproval,
+      approvalDecisionRequired: false,
+    });
+    fixture.detectChanges();
+    const backToListLink: DebugElement = el
+      .queryAll(By.css('a'))
+      .find((button) =>
+        (button.nativeElement as HTMLElement).textContent.includes(
+          'orderApproval.back'
+        )
+      );
+    expect(backToListLink).toBeTruthy();
+
+    const otherButton: DebugElement = el.query(By.css('button'));
```

Comment:
maybe instead of checking for 1 of the other button when the approval decision is false, you can do

```
expect(el.queryAll(By.css('button')).length).toEqual(0);
```

Reply:
This is what the current test does.  In the case where no decision needs to be taken, the minute there his a `<button>` in the html, something is wrong.